src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -0,0 +1,257 @@
+
+            // Switch to the UI thread to update the view.
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            // Unselects the currently selected symbol. This is required in case the current caret position is not in range of any symbols
+            // (symbolToSelect will be null) so that no symbols in the tree are highlighted.
+            if (currentlySelectedSymbol is not null)
+                currentlySelectedSymbol.IsSelected = false;
+
+            if (symbolToSelect is not null)
+                symbolToSelect.IsSelected = true;
+
+            var expansion = expansionOption.First();
+            if (expansion is not ExpansionOption.NoChange)
+                DocumentOutlineHelper.SetIsExpanded(model.DocumentSymbolItems, expansion);
+
+            symbolTree.ItemsSource = model.DocumentSymbolItems;
+        }
+
+        /// <summary>
+        /// Starts a new task to select code when a symbol node is clicked.
+        /// </summary>
+        private void StartJumpToContentTask(DocumentSymbolItem symbol)
+        {
+            _jumpToContentQueue.AddWork(symbol);
+        }
+
+        /// <summary>
+        /// Given a DocumentSymbolItem, moves the caret to its position in the latest active text view.
+        /// </summary>
+        private async ValueTask JumpToContentAsync(ImmutableSegmentedList<DocumentSymbolItem> symbol, CancellationToken cancellationToken)
+        {
+            if (symbol.IsDefault || symbol.IsEmpty)
+                return;
+
+            var model = await _computeUIModelQueue.WaitUntilCurrentBatchCompletesAsync().ConfigureAwait(false);
+            if (model is null)
+                return;
+
+            // Switch to the UI thread to update the latest active text view.
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var activeTextView = GetLastActiveIWpfTextView();
+            if (activeTextView is null)
+                return;
+
+            // When the user clicks on a symbol node in the window, we want to move the cursor to that line in the editor. If we don't
+            // unsubscribe from Caret_PositionChanged first, we will call StartHightlightNodeTask() once we move the cursor ourselves.
+            // This is not ideal because we would be doing extra work to highlight a node that's already highlighted.
+            activeTextView.Caret.PositionChanged -= Caret_PositionChanged;
+
+            // Prevents us from being permanently unsubscribed if an exception is thrown while updating the text view selection.
+            try
+            {
+                // Get the original position of the start of the line of the symbol.
+                var originalPosition = model.OriginalSnapshot.GetLineFromLineNumber(symbol.First().StartPosition.Line).Start.Position;
```

Comment:
i'm not getting the logic here.  we have an LSP Line/Column indicating the location to go to.  but you're only using the Line portion of that Line/Column.  Something seems missing.